experimental/dds/tree2/src/test/feature-libraries/editable-tree/editableTree.binder.spec.ts

```diff
@@ -0,0 +1,1018 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils";
+import { AllowedUpdateType, FieldKey } from "../../../core";
+import {
+	ContextuallyTypedNodeData,
+	getField,
+	BindPath,
+	BindingType,
+	toDownPath,
+	InsertBindingContext,
+	DeleteBindingContext,
+	BindingContext,
+	FlushableDataBinder,
+	BinderOptions,
+	FlushableBinderOptions,
+	createDataBinderBuffering,
+	createFlushableBinderOptions,
+	createDataBinderInvalidating,
+	createDataBinderDirect,
+	InvalidStateBindingContext,
+	DataBinder,
+	InvalidationBinderEvents,
+	OperationBinderEvents,
+	DownPath,
+	BindingContextType,
+	BatchBindingContext,
+	comparePipeline,
+	CompareFunction,
+	compareBinderEventsDeleteFirst,
+	createBinderOptions,
+	BindSyntaxTree,
+	indexSymbol,
+	compileSyntaxTree,
+	BindTree,
+} from "../../../feature-libraries";
+import { brand } from "../../../util";
+import { ISharedTreeView, SharedTreeFactory, ViewEvents } from "../../../shared-tree";
+import { fullSchemaData, personData } from "./mockData";
+
+const fieldAddress: FieldKey = brand("address");
+const fieldZip: FieldKey = brand("zip");
+const fieldStreet: FieldKey = brand("street");
+const fieldPhones: FieldKey = brand("phones");
+const fieldSequencePhones: FieldKey = brand("sequencePhones");
+
+describe("editable-tree: data binder", () => {
+	describe("buffering data binder", () => {
+		it("registers to root, enables autoFlush, matches paths incl. index", () => {
+			const { tree, root, address } = retrieveNodes();
+			const insertTree = compileSyntaxTree({
+				address: {
+					[indexSymbol]: 0,
+					zip: { [indexSymbol]: 1 },
+				},
+			});
```

Comment:
It seems that there is something wrong with the paths reported by the events. Taking this value field as an example: in the `visitDelta`, the `index === 1` for insert, as it follows the delete of this value. Then it comes to the `AnchorSet` and is used just "like that" to build `UpPath.parentIndex`, which is in turn forwarded into `onInsert/onDelete` events of path visitors ([the code in anchorSet](https://github.com/sharptrip/FluidFramework/blob/8b242fdc796714cf1da9ad3f90d02efb122af0c2/experimental/dds/tree2/src/core/tree/anchorSet.ts#L655-L665
)), resulting in the wrong location reported (index 1 instead of index 0 for this node of the value field). I also have tested the same with optional fields - it also reports 1 instead of 0.